lib/node/nodes/gravity.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var Node = require('../node');
+
+var TYPE = 'gravity';
+
+var PARAMS = {
+    source: Node.PARAM.NODE(Node.GEOMETRY.ANY), // "populated places"
+    target: Node.PARAM.NODE(Node.GEOMETRY.POINT), // "malls"
+    weight_column: Node.PARAM.STRING(),
+    weight_threshold: Node.PARAM.NULLABLE(Node.PARAM.NUMBER(),-10e307),
```

Comment:
If you want to have a more complex validation for this param you can do something like:

```diff
diff --git a/lib/node/nodes/gravity.js b/lib/node/nodes/gravity.js
index b9e73fd..0ab615c 100644
--- a/lib/node/nodes/gravity.js
+++ b/lib/node/nodes/gravity.js
@@ -14,7 +14,13 @@ var PARAMS = {
     target_id: Node.PARAM.NUMBER() // the "mall" to be studied
 };

-var Gravity = Node.create(TYPE, PARAMS, {cache: true, version: 1});
+var Gravity = Node.create(TYPE, PARAMS, {cache: true, version: 1,
+    beforeCreate: function (node) {
+        if (node.weight_threshold < 2000) {
+            throw new Error('weight_threshold param should be bigger than 2000');
+        }
+    }
+});

 module.exports = Gravity;
 module.exports.TYPE = TYPE;
```